mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1511,23 +1511,30 @@ def predict_with_latency(X_copy):
             model_predictions = self.dataset.predictions_data
 
         if self.model_type == _ModelType.CLASSIFIER:
-            self.label_list = np.unique(self.y)
-            self.num_classes = len(self.label_list)
-
             if self.predict_fn is not None:
                 self.y_pred = self.predict_fn(self.X.copy_to_avoid_mutation())
             else:
                 self.y_pred = self.dataset.predictions_data
+
+            if self.label_list is None:
+                self.label_list = np.unique(np.concatenate([self.y, self.y_pred]))
+            self.num_classes = len(self.label_list)
             self.is_binomial = self.num_classes <= 2
 
             if self.is_binomial:
-                if self.pos_label in self.label_list:
-                    self.label_list = np.delete(
-                        self.label_list, np.where(self.label_list == self.pos_label)
-                    )
-                    self.label_list = np.append(self.label_list, self.pos_label)
-                elif self.pos_label is None:
+                if self.pos_label is None:
                     self.pos_label = self.label_list[-1]
```

Comment:
This assumes `self.label_list` values are in ascending order, is that ensured ? (e.g., user passed in label_list)

Reply:
Good point! We should sort if user pass in the label list!